pkg/apiserver/rest/dualwriter.go

```diff
@@ -260,3 +295,46 @@ func getName(o runtime.Object) string {
 	}
 	return accessor.GetName()
 }
+
+const dataSyncerInterval = 60 * time.Minute
+
+// StartPeriodicDataSyncer starts a background job that will execute the DataSyncer every 60 minutes
+func StartPeriodicDataSyncer(ctx context.Context, mode DualWriterMode, legacy LegacyStorage, storage Storage,
+	kind string, reg prometheus.Registerer, serverLockService ServerLockService, requestInfo *request.RequestInfo) {
+	klog.Info("Starting periodic data syncer for mode mode: ", mode)
+
+	// run in background
+	go func() {
+		// run it immediately
+		_, _ = runDataSyncer(ctx, mode, legacy, storage, kind, reg, serverLockService, requestInfo)
+
+		ticker := time.NewTicker(dataSyncerInterval)
+		for {
+			select {
+			case <-ticker.C:
+				_, _ = runDataSyncer(ctx, mode, legacy, storage, kind, reg, serverLockService, requestInfo)
+			case <-ctx.Done():
+				return
+			}
+		}
+	}()
+}
+
+// runDataSyncer will ensure that data between legacy storage and unified storage are in sync.
+// The sync implementation depends on the DualWriter mode
+func runDataSyncer(ctx context.Context, mode DualWriterMode, legacy LegacyStorage, storage Storage,
+	kind string, reg prometheus.Registerer, serverLockService ServerLockService, requestInfo *request.RequestInfo) (bool, error) {
+	// ensure that execution takes no longer than necessary
+	const timeout = dataSyncerInterval - time.Minute
+	ctx, cancelFn := context.WithTimeout(ctx, timeout)
+	defer cancelFn()
+
+	// implementation depends on the desired DualWriter mode
+	switch mode {
+	case Mode2:
+		return mode2DataSyncer(ctx, legacy, storage, kind, reg, serverLockService, requestInfo)
+	default:
+		klog.Error("data syncer not implemented for mode mode:", mode)
```

Comment:
Should this be an error or just info/debug?

Reply:
I think we should make it very clear that the implementation is missing, and therefore should be `Info`. I do agree that is not an `error`, though.

I am going to change it to `Info`

Reply:
done!